### DISCUSSION TITLE
Can you explain how the baselineTrim functions?

### DISCUSSION THREAD

**Discussion by dabrahams** [seek-oss/capsize#60] [Q&A] [answered]
I think I understand roughly what's going on with most of the calculations, but the `margin-top` setting on the `::after` pseudo-element is a bit of a mystery to me; if I remove it, that never seems to change anything.

Thanks.
Dave

**Comment by michaeltaranto** [comment, marked as answer]
Best way to look at this is a [quick demo](https://seek-oss.github.io/braid-design-system/playroom/#?code=N4Igxg9gJgpiBcIA8UCWA3ABAZwC4E8AbGAXmGEwAcBDKNAOwHN5MBGABnYBpMBbagB4B1VFFwALFgBZOPfgCdGqeiwA6IdpmoBXXBHWYAvoYB8q%2BpkxJKOAsTIVIhCPLUg0vRvOr4DxsxaWmAAiMLwQmBAAZpgAwtSU2KgAXjCY2klMmPIw1ISYMMS8MPS42JjKSbCRMYkw2lAQ2PDmlkgA9JQBrVZo6AFBvRi2RKTkmABG1GAA1l4Q2vRQboSojOJleeH06jziMGsbLABMmv49gwAKOeioC%2BWFYSW4F1btfd2BbdgwYLh3Fjwowck2mc3kCyWblw2nkAEdtBBUD8-KZXm0%2BuigkD7OQsYMohBSgAxai8VCEfBuABKEAmED0u3xQUJpQAyikYG5WKdKAImV9BkFVvQYAAJA7rXDcgBs7D5AqFQWMWIGSqG6GZllEJHUExghJylx%2BDX0IC1I1xFDQ2EohB80OoE2Iuz41EUygAQgy9Lw3ABadgAOmOAE4AOwAVjCqOZ7TVSolhGczJQGAtOvU1CiuBg8mN9Ua6gtOLG1uRdodmHUuCdLpAcndSnoABUIJQA8GAMzHKShmMgIyGOMJoIdD6vDo-P4Az6DNNYUsgqazeaLZbVkCrKXYLZE137Q7SzCnIcmAByMAEuAKRWe44waof-XMPSQE3kmHjb4-X7nVhsJdxicFw3A8LwfFRUdzwgAB3CoYlgtIcnCdA0gkNIAAMFGbNtKEwzAoghXhMAwzBMOzXN5AI6hymwbRGEYGA8BgZY306f8F1HBdLTLUFVwhdcViPXdCG2A9JSOE8zjRQVLGuGBbnuW8nlKSd3kfV9BSQad-iJXjlzBNcoU3GF4URZEYCgrEF2ZIDgAtVlcFJclKRpOkGTNLhHKJXAOVSbleX5BsLRFcVJOPdRWDlBUQrk5Vh3i0d50xeLBkzEB9UNGAC1NYs0uxOw%2BJtSsqVMusrIbN0PXob1cF9TsQwjaNeFjArvwKpMUwK2yCu1KBdRASi81yotzT6gzxhK%2B0yprCqDHaAAqRsarwxqez7AdMEW9ohxHGyNJfbT2l02ctJS4Z7P48FIQ3dRtw2UTxKqw8pROGSLyvG9HmKUpnyfQ7OMAoqQRA1xN3A7xfEHc5BQAQXKfAFkwMBqEBGB0P2FTfpvCRqBvQlkzg8pkUwegGSobRk1Y9IbD0UisamH4wteSB5ByP5KSDdiunMf7zAbEAMOKbAEAAbXcCAwFFgBdLgQFg0QJFF%2BAxfDGUAA4ZcMIA).

Using html nodes to replicate the pseudo elements, you can see that without the `marginTop` on the `after` the following node is not pulled up to the baseline of the cropped text node.

The display table technique is also being used to prevent margin collapse resulting in unexpected outcomes when using background colours.

Hope that helps

**Reply by dabrahams** [reply]
That's wonderful, thanks.
